apps/server/src/challenges/challenges.controller.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  ApiTags,
+  ApiBearerAuth,
+  ApiCreatedResponse,
+  ApiBadRequestResponse,
+} from '@nestjs/swagger';
+import {
+  Get,
+  Body,
+  Post,
+  Param,
+  Patch,
+  Delete,
+  HttpCode,
+  Controller,
+  HttpStatus,
+} from '@nestjs/common';
+
+import { Roles } from 'src/decorator/roles.decorator';
+
+import { RoleEnum } from '@/common/constants/role.enum';
+
+import {
+  CreateChallengeDto,
+  UpdateChallengeDto,
+} from '@/challenges/dto/challenges.dto';
+
+import { ChallengesService } from '@/challenges/challenges.service';
+
+import { Challenge } from '@/challenges/entities/challenges.entity';
+
+@ApiTags('Challenges')
+@Controller('challenges')
+@ApiBearerAuth()
+export class ChallengesController {
+  constructor(private readonly challengesService: ChallengesService) {}
+
+  @Post()
+  @Roles(RoleEnum.ADMIN)
+  @ApiCreatedResponse({
+    description: 'Created a new challenge',
+    type: Challenge,
+  })
+  @ApiBadRequestResponse({ description: 'Challenge cannot be created' })
+  @HttpCode(HttpStatus.CREATED)
+  public async post(@Body() body: CreateChallengeDto) {
+    return this.challengesService.create(body);
+  }
+
+  @Get()
+  @Roles(RoleEnum.ADMIN, RoleEnum.REVIEWER, RoleEnum.USER)
+  @ApiCreatedResponse({
+    description: 'Fetched all challenges',
+    type: Challenge,
+  })
+  @ApiBadRequestResponse({ description: 'Challenge cannot be fetched' })
+  @HttpCode(HttpStatus.OK)
+  public async getAll() {
+    return this.challengesService.findAll();
+  }
+
+  @Get(':id')
+  @Roles(RoleEnum.ADMIN, RoleEnum.REVIEWER, RoleEnum.USER)
+  @ApiCreatedResponse({
+    description: 'Fetched challenge',
+    type: Challenge,
+  })
+  @ApiBadRequestResponse({ description: 'Challenge cannot be fetched' })
+  @HttpCode(HttpStatus.OK)
+  public async getById(@Param('id') id: number) {
```

Comment:
No need to mark controller methods as `public`

Reply:
Done